nmcli: Introduce IPIP Tunnel

ssahani opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
This work enables to add ipip tunnel via nmcli module

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Feature Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
nmcli
##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes -->
```
ansible 2.8.0.dev0
```

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below
  - nmcli:
       state: present
       type: ipip
       conn_name: ipip_test1
       autoconnect: yes
       ipip_dev: enp0s8
       ipip_local: 192.168.1.2
       ipip_remote: 192.168.1.5
`
```
Signed-off-by: Susant Sahani <susant@redhat.com>

Akasurde commented on lib/ansible/modules/net_tools/nmcli.py:
```diff
@@ -1066,6 +1089,61 @@ def modify_connection_vlan(self):
         return cmd
 
+    def create_connection_ipip(self):
+        cmd = [self.nmcli_bin]
```
You can simply write - 

```python
cmd = [self.nmcli_bin, 'con', 'add', 'type', 'ip-tunnel', 'mode', 'ipip', 'con-name']
```

Akasurde commented on lib/ansible/modules/net_tools/nmcli.py:
```diff
@@ -1066,6 +1089,61 @@ def modify_connection_vlan(self):
+        return cmd
+
+    def modify_connection_ipip(self):
+        cmd = [self.nmcli_bin]
```
same as above.

```python
cmd = [self.nmcli_bin, 'con', 'mod']
```

Akasurde reviewed (changes requested):
Thanks for the contribution. Could you please write a test for this change here - https://github.com/ansible/ansible/blob/devel/test/units/modules/net_tools/test_nmcli.py ? Thanks.

ssahani commented:
@Akasurde  Ping

Akasurde commented:
@ssahani Could you please rebase this ? needs_info

ssahani commented:
Yes  I am doing this right now for SIT. Will update. thanks !

ssahani committed:
nmcli: Introduce IPIP Tunnel

This work enables to add ipip tunnel via nmcli module

```
  - nmcli:
       state: present
       type: ipip
       conn_name: ipip_test1
       autoconnect: yes
       ip_tunnel_dev: enp0s8
       ip_tunnel_local: 192.168.1.2
       ip_tunnel_remote: 192.168.1.5
```

version2: Added tests
Signed-off-by: Susant Sahani <susant@redhat.com>

Rebase

ssahani commented:
Rebased and pushed.

Akasurde commented:
rebuild_merge
